### DISCUSSION TITLE
Request to add Hillside 52 support

### DISCUSSION THREAD

**Discussion by burgessa23** [manna-harbour/miryoku#230] [Q&A] [answered]
I see it on the out-of-tree list, but it's unchecked. Is this something I can do myself in my fork, or does it need to be done in the main repo? 

The shield config is: https://github.com/mmccoyd/zmk-config/tree/main/config/boards/shields/hillside52

Thanks!

**Comment by manna-harbour** [comment, marked as answer]
The process of adding support for new keyboards isn't documented, but if you'd like to do it yourself there are plenty of examples in the commit history. Otherwise I'm happy to do it.

You can add it only in your fork, but if you do, submit a PR so that others can use it too.

For the mapping file, it's probably easiest to copy the hillside48 file and add the extra keys. The thumb combo lines can be removed.

**Reply by burgessa23** [reply]
I'll give it a go starting from the 46 and post back. Thanks

**Comment by burgessa23** [comment]
A couple of questions... 

1. Where do I specify the shield additions for my screens? e.g. my build.yaml has:

``` 
 - board: nice_nano_v2
    shield: hillside52_left nice_view_adapter nice_view
  - board: nice_nano_v2
    shield: hillside52_right nice_view_adapter nice_view
```

2. Can I add additional items to the keymap file (behaviour triggers, etc.) for the key positions not used by the miryoku layout?

**Reply by manna-harbour** [reply]
1. You don't need to do anything when adding support for the shield. After that you can specify multiple shields with the workflow shield option. Use the `Build Inputs` workflow, or see the `Build Example corne nice_nano_v2 nice_view` build example workflow.

**Reply by manna-harbour** [reply]
2. If you'll be submitting a PR, then leave extra keys as none, and in your fork you can customise the mapping with https://github.com/manna-harbour/miryoku/discussions/85. If not you can customise the mapping directly.

**Reply by axburgess-godaddy** [reply]
Thanks, is there an example of a a customization that uses leftover keys not in the default layout?

**Reply by manna-harbour** [reply]
This is one of the examples linked from the above mentioned customisation docs: https://github.com/LilRooks/miryoku_zmk/blob/335945efad01125e64d54ccbe15a63c96ce299ba/miryoku/custom_config.h#L23-L33.

**Comment by axburgess-godaddy** [comment]
Thanks! If I'm having build issues, where's the best place to ask questions?

**Reply by manna-harbour** [reply]
You can link your repo here, or contact me privately.
https://github.com/manna-harbour/manna-harbour/tree/main/contact

**Reply by burgessa23** [reply]
https://github.com/burgessa23/miryoku_zmk/actions/runs/5674134192/job/15377115625#step:6:579 is the error I keep getting whenever adding the nice!view and adapter shields. sent you a discord request btw
